components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -599,6 +599,27 @@ export class TypeORMUserDBImpl implements UserDB {
             .getCount();
         return blockedUsers > 0;
     }
+
+    async findOrganizationalUser(organizationId: string, email: string): Promise<MaybeUser> {
```

Comment:
Would be great to find a common name for the "user who is owned by the Org". Sven used "owned" as the term here https://github.com/gitpod-io/gitpod/pull/17538/files#diff-4c72608b2e3eb106601a9bd44fbb13c2c551329302994c01dcab2586f68192d0R46 so in this case, I'd name it

```suggestion
    async findOwnedUsers(organizationId: string, email: string): Promise<MaybeUser> {
```

or 
```suggestion
    async getUser(organizationId: string, email: string, owned: bool): Promise<MaybeUser> {
```
here, the signature and types communicate the intent such that we don't need to push that into the method description.

2. Do we need to guarantee that the user is not only owned by the Org, but also that they have a relationship with the org through the TeamMembership table?

Reply:
renamed.